Patch taken from GoboLinux:

The version of the function "futimens" used by Gzip is incompatible with the
version that Glibc 2.7+ provides. Prefixing the Gzip function with gz_

Signed-off-by: Jonas Karlsson <jonas@gobolinux.org>

--- gzip.c.orig
+++ gzip.c
@@ -1637,7 +1642,7 @@
 	}
       }
 
-    if (futimens (ofd, ofname, timespec) != 0)
+    if (gz_futimens (ofd, ofname, timespec) != 0)
       {
 	int e = errno;
 	WARN ((stderr, "%s: ", program_name));
--- lib/utimens.h.orig
+++ lib/utimens.h
@@ -1,3 +1,3 @@
 #include <time.h>
-int futimens (int, char const *, struct timespec const [2]);
+int gz_futimens (int, char const *, struct timespec const [2]);
 int utimens (char const *, struct timespec const [2]);
--- lib/utimens.c.orig
+++ lib/utimens.c
@@ -75,7 +75,7 @@
    Return 0 on success, -1 (setting errno) on failure.  */
 
 int
-futimens (int fd ATTRIBUTE_UNUSED,
+gz_futimens (int fd ATTRIBUTE_UNUSED,
 	  char const *file, struct timespec const timespec[2])
 {
   /* Some Linux-based NFS clients are buggy, and mishandle time stamps
@@ -185,5 +185,5 @@
 int
 utimens (char const *file, struct timespec const timespec[2])
 {
-  return futimens (-1, file, timespec);
+  return gz_futimens (-1, file, timespec);
 }

